integration_tests/src/main/python/asserts.py

```diff
@@ -459,37 +459,39 @@ def do_it_all(spark):
 def _assert_gpu_and_cpu_are_equal(func,
     mode,
     conf={},
-    is_cpu_first=True):
+    is_cpu_first=True,
+    result_converter_func_before_compare=None):
     (bring_back, collect_type) = _prep_func_for_compare(func, mode)
     conf = _prep_incompat_conf(conf)
 
     def run_on_cpu():
         print('### CPU RUN ###')
         global cpu_start
         cpu_start = time.time()
-        global from_cpu
         from_cpu = with_cpu_session(bring_back, conf=conf)
         global cpu_end
         cpu_end = time.time()
+        return from_cpu
 
     def run_on_gpu():
         print('### GPU RUN ###')
         global gpu_start
         gpu_start = time.time()
-        global from_gpu
         from_gpu = with_gpu_session(bring_back, conf=conf)
         global gpu_end
         gpu_end = time.time()
+        return from_gpu
 
-    if is_cpu_first:
-        run_on_cpu()
-        run_on_gpu()
-    else:
-        run_on_gpu()
-        run_on_cpu()
-
+    (from_cpu, from_gpu) = (run_on_cpu(), run_on_gpu()) if is_cpu_first else (run_on_gpu(), run_on_cpu())
```

Comment:
This is a little confusing to me.  The meaning of `is_cpu_first` used to be the choice of whether to execute the CPU run first.

On this line, it looks like if `is_cpu_first == False`, the GPU run happens first, but the result is assigned to `from_cpu`.  Am I reading this incorrectly?

Reply:
Hmm, same question since the order matters in ```result_converter_func_before_compare```.

Reply:
Good catch